src/main/scala/com.socrata.tileserver/services/TileService.scala

```diff
@@ -227,13 +228,40 @@ object TileService {
     val dis = rs.open(new DataInputStream(resp.inputStream(Long.MaxValue)))
 
     try {
-      val headers = MsgPack.unpack(dis, MsgPack.UNPACK_RAW_AS_STRING).asInstanceOf[Map[String, Any]]
-      val jsonHeaders = JObject(headers.mapValues(v => JString(v.toString)))
-      headers.asInt("geometry_index") match {
-        case geomIndex: Any if geomIndex < 0 =>
-          Failure(InvalidSoqlPackException(headers))
-        case geomIndex: Any =>
-          Success(jsonHeaders -> new FeatureJsonIterator(dis, geomIndex))
+      val soqlIter = new SoQLPackIterator(dis)
+      val jsonHeaders = JObject(soqlIter.headers.mapValues(v => JString(v.toString)))
+      if (soqlIter.geomIndex < 0) {
+        Failure(InvalidSoqlPackException(soqlIter.headers))
+      } else {
+        val colNames = soqlIter.schema.map(_._1).toArray
+        val featureJsonIter = soqlIter.map { case soqlRow =>
```

Comment:
This logic seems like it might be somewhat general/generalizable.  I'd love to see this pushed into the library.  (Though the code itself looks reasonable.)

Maybe with syntax like:
```scala
soqlIter.map(_.getGeometry)
```